
Diagnostics:
error: cannot mutably borrow since mutable references exist
   ┌─ tests/reference-safety/v1-borrow-tests/imm_borrow_on_mut_invalid.move:25:29
   │
24 │         let field_ref = set_and_pick(account, copy point_ref);
   │                         ------------------------------------- previous mutable call result
25 │         let x_val = *freeze(&mut point_ref.x);
   │                     --------^^^^^^^^^^^^^^^^-
   │                     │       │
   │                     │       mutable borrow attempted here
   │                     requirement enforced here
26 │         let returned_ref = bump_and_give(field_ref);
   │                            ------------------------ conflicting reference `field_ref` used here

error: cannot freeze value  since multiple mutable references exist
   ┌─ tests/reference-safety/v1-borrow-tests/imm_borrow_on_mut_invalid.move:25:22
   │
24 │         let field_ref = set_and_pick(account, copy point_ref);
   │                         ------------------------------------- conflicting mutable call result
25 │         let x_val = *freeze(&mut point_ref.x);
   │                      ^^^^^^^^^^^^^^^^^^^^^^^^
   │                      │      │
   │                      │      originating mutable field borrow
   │                      frozen here

error: cannot immutably borrow since mutable references exist
   ┌─ tests/reference-safety/v1-borrow-tests/imm_borrow_on_mut_invalid.move:39:23
   │
38 │         let field_ref = set_and_pick(account, copy point_ref);
   │                         ------------------------------------- previous mutable call result
39 │         let x_val = *&freeze(point_ref).x;
   │                     --^^^^^^^^^^^^^^^^^--
   │                     │ │
   │                     │ immutable borrow attempted here
   │                     requirement enforced here
40 │         let returned_ref = bump_and_give(field_ref);
   │                            ------------------------ conflicting reference `field_ref` used here
